### PR TITLE
Implemented TLS client certificate authentication

### PR DESCRIPTION
**Pull request by Fluepke**
### Proposed changes
Implemented options to specify:
* a client certificate (and key) in *PEM* format to authenticate against the server
* a CA certificate in *PEM* format to authenticate the server

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-prometheus-exporter/blob/master/CONTRIBUTING.md) guide
- [x] I have proven my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have ensured the README is up to date
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch on my own fork

### CODE REVIEW

**Comment by pleshakov** [comment]
@Fluepke thx for the PR! we'll review it and get back to you shortly

File: exporter.go [hunk 1]
```diff
@@ -289,8 +303,27 @@ func main() {
...
 	registry.MustRegister(buildInfoMetric)
 
+	sslConfig := &tls.Config{InsecureSkipVerify: !*sslVerify}
+	if *sslCaCert != "" {
+		caCert, err := ioutil.ReadFile(*sslCaCert)
+		if err != nil {
+			log.Fatalf("Loading CA cert failed: %v", err)
+		}
+		sslCaCertPool := x509.NewCertPool()
+		sslCaCertPool.AppendCertsFromPEM(caCert)
```
Comment (pleshakov) [inline exporter.go]: `AppendCertsFromPEM` function reports whether any certificates were successfully parsed in a boolean returned argument. Could you possible check that result and fatal with an error if there is a problem?

File: exporter.go [hunk 2]
```diff
@@ -247,6 +252,15 @@ For NGINX, the stub_status page must be available through the URI. For NGINX Plu
...
 		"Perform SSL certificate verification. The default value can be overwritten by SSL_VERIFY environment variable.")
+	sslCaCert = flag.String("nginx.ssl-ca-cert",
+		defaultSslCaCert,
+		"The CA certificate to use to validate the servers SSL certificate. The default value can be overwritten by SSL_CA_CERT environment variable.")
+	sslClientCert = flag.String("nginx.ssl-client-cert",
+		defaultSslClientCert,
+		"The client certificate to use when connecting to the server. The default value can be overwritten by SSL_CLIENT_CERT environment variable.")
+	sslClientKey = flag.String("nginx.ssl-client-key",
+		defaultSslClientKey,
+		"The client certificate key to use when connecting to the server. The default value can be overwritten by SSL_CLIENT_KEY environment variable.")
```
Comment (lucacome) [inline exporter.go]: Maybe it's better to say something like "Path to the client certificate file.."? To make it clearer that you need to pass the file path.

**Comment by lucacome** [comment]
Hi @Fluepke 

just wanted to check if you're planning to address our suggestions or if you want us to take over the PR 🙂 

Thanks

**Pull request merged**